src/pdf/ShoppingList/databaseFetch.ts

```diff
@@ -0,0 +1,43 @@
+import supabase, { Schema } from "@/supabase";
+
+export const fetchParcels = async (parcelID: string): Promise<Schema["parcels"]> => {
+    const { data, error } = await supabase.from("parcels").select().eq("primary_key", parcelID);
+    if (error !== null) {
+        throw Error(`${error.code}: ${error.message}`);
+    }
+    if (data.length !== 1) {
+        const errorMessage =
+            (data.length === 0 ? "No " : "Multiple ") + "records match this parcel ID.";
+        throw Error(errorMessage);
+    }
+    return data[0];
+};
+
+export const fetchClients = async (primaryKey: string): Promise<Schema["clients"]> => {
+    const { data, error } = await supabase.from("clients").select().eq("primary_key", primaryKey);
+    if (error !== null) {
+        throw Error(`${error.code}: ${error.message}`);
+    }
+    if (data.length !== 1) {
+        const errorMessage =
+            (data.length === 0 ? "No " : "Multiple ") + "records match this client ID.";
+        throw Error(errorMessage);
+    }
+    return data[0];
+};
+
+export const fetchFamilies = async (familyID: string): Promise<Schema["families"][]> => {
+    const { data, error } = await supabase.from("families").select().eq("family_id", familyID);
+    if (error !== null) {
+        throw Error(`${error.code}: ${error.message}`);
```

Comment:
assuming it's safe to show the client the error message?

Reply:
Miss spelling the table name to fam gives the following code and message: Error: 42P01: relation "public.fam" does not exist.

We will probably need to test this for other failed cases as well. Or to be safe and user friendly, perhaps we have a global error message which maps the error code to a "user friendly" error message eg. 4xx: "There was a bad request."

However, I will leave it as is right now as I think this should be done in another ticket across the whole project (VFB-22).

Reply:
can we remove error.message for now, and re-add it back in vfb-22?